paper/paper.md

```diff
@@ -0,0 +1,103 @@
+The Causal Testing Framework is written in python but is language agnostic in terms of the system under test.
+All that is required is a set of properties to be validated, a causal model, and a set of software runtime data.
+
+# Causal Testing
+Causal Testing [@clark2023testing] has four main steps, outlined in \ref{fig:schematic}.
+Firstly, the user supplies a causal model, which takes the form of a directed acyclic graph (DAG) in which an edge $X \to Y$ represents variable $X$ having a direct causal effect on variable $Y$.
+Secondly, the user supplies a set of causal properties to be tested.
+Such properties can be generated from the causal DAG [@clark2023metamorphic]: for each $X \to Y$ edge, a test to validate the presence of a causal effect is generated, and for each missing edge, a test to validate independence is generated.
+The user may also refine tests to validate the nature of a particular relationship.
+Next, the user supplies a set of runtime data in the form of a table with each column representing a variable and rows containing the value of each variable for a particular run of the software.
+Finally, the Causal Testing Framework automatically validates the supplied causal properties by using the supplied causal DAG and data to calculate a causal effect estimate, and validating this against the expected causal relationship.
+
+![Causal Testing workflow.\label{fig:schematic}](../images/schematic.png)
+
+## Test Adequacy
+Because the properties being tested are completely separate from the data used to validate them, traditional coverage-based metrics are not appropriate here.
+The Causal Testing Framework instead evaluates the adequacy of a particular dataset by calculating a statistical metric [@foster2024adequacy] based on the stability of the causal effect estimate, with numbers closer to zero representing more adequate data.
+
+## Missing Variables
+Causal Testing works by using the supplied causal DAG to identify those variables which need to be statistically controlled for to remove their biassing effect on the causal estimate.
+This typically means we need to know their values.
+However, the Causal Testing Framework can still sometimes estimate unbiased causal effects using Instrumental Variables, an advanced Causal Inference technique.
+
+## Feedback Over Time
+Many scientific models involve iterating several interacting processes over time.
+These processes often feed into each other, and can create feedback cycles.
+Traditional Causal Inference cannot handle this, however the Causal Testing Framework uses another advanced Causal Inference technique, g-methods, to enable the estimation of causal effects even when there are feedback cycles between variables.
```

Comment:
I would add "family of" when referring to g-methods - i.e.:

"Traditional Causal Inference cannot handle this, however the Causal Testing Framework uses another **family of** advanced Causal Inference technique, g-methods, to enable the estimation of causal effects even when there are feedback cycles between variables."